File: CSharpAthletic/Persistence/EntryRepositoryDB.cs
using System;
using System.Collections.Generic;
using System.Data;
using Domain;
using Domain.DTO;
using Persistence.DB;

namespace Persistence
{
    public class EntryRepositoryDB : IEntryRepository
    {
        public EntryRepositoryDB()
        {
        }

        public IEnumerable<Entry> GetAll()
        {
            return null;
        }

        public Entry GetById(long id)
        {
            return null;
        }

        public IEnumerable<EventCountDTO> GetEventsNumber()
        {
            IDbConnection con = DBUtils.getConnection();
            IList<EventCountDTO> events = new List<EventCountDTO>();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "SELECT Event, AgeGroup, COUNT(*) AS Number FROM (SELECT Event1 AS Event, AgeGroup FROM Entry UNION ALL SELECT Event2, AgeGroup FROM Entry WHERE Event2 != 'NONE') GROUP BY Event, AgeGroup;";

                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        Event Eventt = (Event)Enum.Parse(typeof(Event), dataR.GetString(0));
                        AgeGroup ageGroup = (AgeGroup)Enum.Parse(typeof(AgeGroup), dataR.GetString(1));
                        long Count = dataR.GetInt64(2);
                        EventCountDTO eventt = new EventCountDTO { Event = Eventt, AgeGroup = ageGroup, Count = Count };
                        events.Add(eventt);
                    }
                }
            }
            return events;
        }

        public long Save(Entry entry)
        {
            IDbConnection con = DBUtils.getConnection();

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "INSERT INTO Entry (IDChild, Event1, Event2, AgeGroup) VALUES (@idChild, @event1, @event2, @ageGroup);";
                IDbDataParameter paramIdChild = comm.CreateParameter();
                paramIdChild.ParameterName = "@idChild";
                paramIdChild.Value = entry.Child.Id.ToString();
                comm.Parameters.Add(paramIdChild);

                IDbDataParameter paramEvent1 = comm.CreateParameter();
                paramEvent1.ParameterName = "@event1";
                paramEvent1.Value = entry.Event1.ToString();
                comm.Parameters.Add(paramEvent1);

                IDbDataParameter paramEvent2 = comm.CreateParameter();
                paramEvent2.ParameterName = "@event2";
                paramEvent2.Value = entry.Event2.ToString();
                comm.Parameters.Add(paramEvent2);

                IDbDataParameter paramAgeGroup = comm.CreateParameter();
                paramAgeGroup.ParameterName = "@ageGroup";
                paramAgeGroup.Value = entry.AgeGroup.ToString();
                comm.Parameters.Add(paramAgeGroup);

                var result = comm.ExecuteNonQuery();
                if (result == 0)
                    throw new Exception("No task added!");

                return 0;
            }
        }
    }
}

File: JavaAthletic/src/main/java/repo/EntryRepositoryDB.java
package repo;

import domain.AgeGroup;
import domain.Entry;
import domain.Event;
import dtos.ChildNoEventsDTO;
import dtos.EventCountDTO;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import utils.DBConnection;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class EntryRepositoryDB implements IEntryRepository {
    private final DBConnection dbUtils;
    private static final Logger logger = LogManager.getLogger();

    public EntryRepositoryDB(Properties props){
        logger.info("Initializing EntryRepositoryDB with properties: {} ", props);
        dbUtils = new DBConnection(props);
    }

    @Override
    public Long save(Entry entry) {
        logger.traceEntry("Saving task {} ", entry);
        Connection con = dbUtils.getConnection();
        try(PreparedStatement preStmt = con.prepareStatement("INSERT INTO Entry (IDChild, Event1, Event2, AgeGroup) VALUES (?, ?, ?, ?);")){
            preStmt.setLong(1, entry.getChild().getId());
            preStmt.setString(2, entry.getEvent1().toString());
            preStmt.setString(3, entry.getEvent2().toString());
            preStmt.setString(4, entry.getAgeGroup().toString());
            int result = preStmt.executeUpdate();
            logger.trace("Saved {} instances", result);
        } catch (SQLException ex){
            logger.error(ex);
            System.err.println("Error DB " + ex);
        }
        logger.traceExit();
        return null;
    }

    @Override
    public List<Entry> getAll() {
        return null;
    }

    @Override
    public Entry getById(Long aLong) {
        return null;
    }

    @Override
    public List<EventCountDTO> getEventsNumber() {
        /*
        Returneaza lista cu probele si cati participanti s-au inscrisi la fiecare proba
         */
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        List<EventCountDTO> events = new ArrayList<>();
        try(PreparedStatement preStmt = con.prepareStatement("SELECT Event, AgeGroup, COUNT(*) AS Number FROM (SELECT Event1 AS Event, AgeGroup FROM Entry UNION ALL SELECT Event2, AgeGroup FROM Entry WHERE Event2 != 'NONE') GROUP BY Event, AgeGroup;")){
            try(ResultSet result = preStmt.executeQuery()){
                while(result.next()){
                    Event event = Event.valueOf(result.getString("Event"));
                    AgeGroup ageGroup = AgeGroup.valueOf(result.getString("AgeGroup"));
                    long number = result.getLong("Number");
                    EventCountDTO eventDTO = new EventCountDTO(event, ageGroup, number);
                    events.add(eventDTO);
                }
            }
        } catch(SQLException e){
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        logger.traceExit(events);
        return events;
    }


}
